lib/sinatra/base.rb

```diff
@@ -181,7 +181,7 @@ def drop_body?
   class ExtendedRack < Struct.new(:app)
     def call(env)
       result, callback = app.call(env), env['async.callback']
-      return result unless callback and async?(*result)
+      return result unless callback && async?(*result)
```

Comment:
Ok, wait. I think I'm wrong. `unless` suggests that this is intended to be read as a boolean expression.